helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -448,36 +533,42 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
         version = Optional.of(node.getStat().getVersion());
         curHosts = Json.read(node.getBytes(), new TypeReference<List<String>>() {});
       } catch (JsonMappingException e) {
-        curHosts = Collections.emptyList();
+        curHosts = emptyList();
       } catch (NoNodeException e) {
         // DG was deleted -- abort.
         return;
       }
 
       if (!version.isPresent() || !hosts.equals(curHosts)) {
         // Node not present or hosts have changed
-        final List<ZooKeeperOperation> ops = Lists.newArrayList();
-        ops.add(set(Paths.statusDeploymentGroupHosts(groupName), Json.asBytes(hosts)));
-
-        final DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
-        ImmutableList<Map<String, Object>> events = ImmutableList.of();
-
-        if (deploymentGroup.getJobId() != null) {
-          final DeploymentGroupStatus deploymentGroupStatus = getDeploymentGroupStatus(groupName);
-          if (deploymentGroupStatus == null || deploymentGroupStatus.getState() != FAILED) {
-            final RollingUpdateOp op =
-                getInitRollingUpdateOps(deploymentGroup, hosts, HOSTS_CHANGED, client);
-            ops.addAll(op.operations());
-            events = op.events();
-          }
+        final List<ZooKeeperOperation> operations = Lists.newArrayList();
+        operations.add(set(Paths.statusDeploymentGroupHosts(groupName), Json.asBytes(hosts)));
+
+        final RollingOperation lastOp = getLastRollingOperation(client, groupName);
+        if (lastOp == null) {
+          // This deployment group has never been deployed to, so don't deploy to the new hosts.
+          log.info("starting zookeeper transaction for updateDeploymentGroupHosts on "
+                   + "deployment-group name {}. List of operations: {}",
+                   groupName, operations);
+          client.transaction(operations);
```

Comment:
Isn't that still the case?

My understanding is that previously it would hit `client.getNode(Paths.statusDeploymentGroupHosts(groupName))`, raise a `NoNodeException`, and thus `return` if the DG didn't exist. That should still be the case.

At the line you've commented on it's handling the case where the deployment group exists, but has never been deployed to before. So `client.transaction(operations)` should only be committing the operation that adds the updated hots.

What am I missing?